### PR TITLE
Story/expand side panel/822

### PR DESCRIPTION
**Pull request by conbrad**
https://wps-pr-922.apps.silver.devops.gov.bc.ca/morecast

- Adds `ExpandableContainer` component that holds the `SidePanel`
- Refactors `SidePanel` to remove concept of open/close and adds it to `ExpandableContainer` which contains the expand/collapse/close controls and functionality
- Updates `npm` packages with `yarn upgrade`

Collapsed:
Screen Shot 2021-04-23 at 2 45 46 PM

Expanded:
Screen Shot 2021-04-23 at 11 51 58 AM

### CODE REVIEW

**Review by andrea-williams — APPROVED** [review]
One relative file import to be fixed (please), but otherwise I'm happy.
I'd recommend that we get Ambrose to take a look at it first before merging though.

File: web/src/features/fireWeather/pages/MoreCastPage.tsx [hunk 1]
```diff
@@ -22,6 +22,8 @@ import SidePanel from 'features/fireWeather/components/SidePanel'
 import NetworkErrorMessages from 'features/fireWeather/components/NetworkErrorMessages'
 import WeatherMap from 'features/fireWeather/components/maps/WeatherMap'
 import { getStations } from 'api/stationAPI'
+import ExpandableContainer from '../components/ExpandableContainer'
```
[the remark this answers is not in the document]
Comment (conbrad) [reply web/src/features/fireWeather/pages/MoreCastPage.tsx:25]: Just found typescript setting in vscode to prefer non-relative imports, will fix

**Comment by Sybrand** [comment]
@conbrad I'm just worried about this scenario:
image

File: web/src/features/fireWeather/pages/MoreCastPage.tsx [hunk 2]
```diff
@@ -68,7 +70,21 @@ const MoreCastPage = () => {
...
+  const [sidePanelWidth, setSidePanelWidth] = useState(
+    shouldInitiallyShowSidePanel ? PARTIAL_WIDTH : 0
+  )
+
+  const [mapCenter, setMapCenter] = useState(CENTER_OF_BC)
+  const expandSidePanel = () => setSidePanelWidth(FULL_WIDTH)
+  const collapseSidePanel = () => setSidePanelWidth(PARTIAL_WIDTH)
+
+  const setNewMapCenter = (newMapCenter: number[]) => {
+    setMapCenter(newMapCenter)
```
Comment (Sybrand) [inline web/src/features/fireWeather/pages/MoreCastPage.tsx:82]: Maybe some comments around why this is needed?

File: web/src/utils/constants.ts [hunk 3]
```diff
@@ -17,3 +17,10 @@ export const FIRE_WEATHER_ROUTE = '/fire-weather'
 export const MORECAST_ROUTE = '/morecast'
 export const HFI_CALC_ROUTE = '/hfi-calculator'
 export const C_HAINES_ROUTE = '/c-haines'
+
+// ExpandableContainer widths
+export const PARTIAL_WIDTH = 850
+export const FULL_WIDTH = 1200
```
Comment (Sybrand) [inline web/src/utils/constants.ts:23]: I've been wondering about this - why can't we just go 100% once you've fully expanded?

**Review by Sybrand — APPROVED** [review]
Two comments - up to you - looks good and works.

File: web/src/utils/constants.ts (hunk 3 above)
Comment (conbrad) [reply web/src/utils/constants.ts:23]: It does -- I just need to use some number because state is represented as a number

Comment (conbrad) [reply web/src/utils/constants.ts:23]: https://github.com/bcgov/wps/pull/922/files#diff-6e18a3c8f2248cba5aab52d282913a1a4080319755220ba5fe852038282fc202R10

File: web/src/stories/WeatherMap.stories.tsx [hunk 4]
```diff
@@ -15,7 +16,7 @@ type PropTypes = React.ComponentProps<typeof WeatherMap>
 const Template: Story<PropTypes> = () => (
   <div style={{ height: '500px' }}>
     <Provider store={store}>
-      <WeatherMap />
+      {<WeatherMap center={CENTER_OF_BC} isCollapsed={false} setMapCenter={_ => {}} />}
```
Comment (Sybrand) [inline web/src/stories/WeatherMap.stories.tsx:19]: Oh - actually - could you please resolve the sonarcloud code smell associated with this?

File: web/src/features/fireWeather/pages/MoreCastPage.tsx (hunk 2 above)
Comment (conbrad) [reply web/src/features/fireWeather/pages/MoreCastPage.tsx:84]: Added clarifying comment

**Review by andrea-williams — APPROVED** [review]

**Pull request merged**